### PR TITLE
Workaround snapd apparmor profiles not loading on boot

### PR DESCRIPTION
**Pull request by Cynerva**
Hacky workaround for https://github.com/conjure-up/conjure-up/issues/1448

### CODE REVIEW

**Comment by Cynerva** [comment]
@battlemidget could you review this?
@kwmonroe can you as well? This is pretty hacky so I would appreciate the extra eyes on it.

**Review by adam-stokes — APPROVED** [review]
lgtm

**Comment by ktsakalozos** [comment]
lgtm, although sad we have to do this.

File: templates/cdk-service-kicker [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 set -eu
 
-# This service runs for a few minutes and restarts any failing CDK services
-# it sees during that time.
-#
-# This is a hacky workaround for this bug:
-# https://github.com/juju-solutions/bundle-canonical-kubernetes/issues/357
+# This service runs on boot to work around issues relating to LXD and snapd.
 
+# Workaround for https://github.com/conjure-up/conjure-up/issues/1448
```
Comment (kwmonroe) [inline templates/cdk-service-kicker:6]: Can we wrap this so it only applies to lxd hosts? I'd like to limit our hacks to the envs that need hacking.

File: templates/cdk-service-kicker [hunk 2]
```diff
@@ -4,11 +4,15 @@ set -eu
 # This service runs on boot to work around issues relating to LXD and snapd.
 
 # Workaround for https://github.com/conjure-up/conjure-up/issues/1448
-(set +e
-  echo "applying snapd apparmor profiles"
-  apparmor_parser /var/lib/snapd/apparmor/profiles/*
-  echo "apparmor_parser: exit status $?"
-)
+if [ -f '/proc/1/environ' ] && grep '^container=lxc' /proc/1/environ; then
```
Comment (kwmonroe) [inline templates/cdk-service-kicker]: Super small nit pick, but there's some gobbledeegook output with grepping the (aparently) binary environ file:
```
root@juju-client-box-1:~# if [ -f '/proc/1/environ' ] && grep '^container=lxc' /proc/1/environ; then echo CONTAINERED; fi
Binary file /proc/1/environ matches
CONTAINERED
```

If you feel like it, get rid of the extra output with a `-q` to grep:
```
root@juju-client-box-1:~# if [ -f '/proc/1/environ' ] && grep -q '^container=lxc' /proc/1/environ; then echo CONTAINERED; fi
CONTAINERED
```

**Review by kwmonroe — APPROVED** [review]
This addresses my concern to limit the hack to containers.  Small suggestion inline, but +1 overall.

File: templates/cdk-service-kicker (hunk 2 above)
Comment (Cynerva) [reply templates/cdk-service-kicker]: I like the gobbledeegook. It gives it that special half-assed flavor that goes well with a service like cdk-service-kicker.

I'll add the `-q`, thanks :+1:

**Comment by Cynerva** [comment]
Added the `-q`, manually tested on LXD and non-LXD and all seems well. Ready to merge if you have no further concerns :+1:

**Pull request merged**